apps/gatsby/src/Sidebar.js

```diff
@@ -34,16 +35,57 @@ export default class Sidebar extends React.Component {
 
     const content = this.props.sdk.entry.getSys();
     this.setManifestId(content);
+    this.setLastPublishedDateTime(content);
   }
 
+  /**
+   * manifestId is used by the Gatsby Content Sync preview feature to match states of content
+   * in Contentful with preview builds on Gatsby Cloud
+   */
   setManifestId = (content) => {
     const { id, space, updatedAt } = content;
     const manifestId = `${space.sys.id}-${id}-${updatedAt}`;
     this.setState({ manifestId });
   }
 
+  setLastPublishedDateTime = (content) => {
+    this.setState({
+      lastPublishedDateTime: content.publishedAt,
+    });
+  }
+
+  /**
+   * lastPublishedDateTime is used to track when the built in Contentful publish button is clicked
+   * and then kick off production builds of the Gatsby site accordingly
+   */
+  maybeStartProductionBuild = (content) => {
+    const { webhookUrl, authToken } = this.sdk.parameters.installation;
+    const { lastPublishedDateTime } = this.state;
+
+    console.log('maybe making a build....', { lastPublishedDateTime, state: this.state });
+
+    // if these timestamps are equal than the content has not been published
+    if (!lastPublishedDateTime || lastPublishedDateTime === content.publishedAt) {
```

Comment:
i might be a bit confused- there is no chance the ts would vary (its only created on one place in the cli, and is essentially an identified for that built?

Reply:
The timestamp exists on the `Content`'s metadata and then metadata changes will trigger a change to `onSysChanged` which will return the `content` object that is used here. The timestamp in this case is stored and maintained by Contentful

Does that answer your Q?

Reply:
Just updated that comment as well to help add some more clarity to future maintainers!